java/json/src/main/java/org/bondlib/SimpleJsonReader.java

```diff
@@ -0,0 +1,480 @@
+        final JsonToken currentToken = this.advanceParserToNextToken();
+        if (currentToken != JsonToken.VALUE_NUMBER_FLOAT && currentToken != JsonToken.VALUE_NUMBER_INT) {
+            throw createInvalidBondDataException(
+                    "Current JSON token is '%s' where either '%s' or '%s' for Bond 'float' data type is expected.",
+                    currentToken,
+                    JsonToken.VALUE_NUMBER_FLOAT,
+                    JsonToken.VALUE_NUMBER_INT);
+        }
+        // rely on Jackson to parse the current token as a float
+        try {
+            final float valueAsFloat = this.parser.getFloatValue();
+            return valueAsFloat;
+        } catch (JsonParseException e) {
+            throw createInvalidBondDataException(
+                    e,
+                    "Current JSON float value '%s' is outside of bounds for the expected Bond data type 'float'.",
+                    this.parser.getValueAsString());
+        }
+    }
+
+    @Override
+    public final double readDouble() throws IOException {
+        final JsonToken currentToken = this.advanceParserToNextToken();
+        if (currentToken != JsonToken.VALUE_NUMBER_FLOAT && currentToken != JsonToken.VALUE_NUMBER_INT) {
+            throw createInvalidBondDataException(
+                    "Current JSON token is '%s' where either '%s' or '%s' for Bond 'double' data type is expected.",
+                    currentToken,
+                    JsonToken.VALUE_NUMBER_FLOAT,
+                    JsonToken.VALUE_NUMBER_INT);
+        }
+        // rely on Jackson to parse the current token as a double
+        try {
+            final double valueAsDouble = this.parser.getDoubleValue();
+            return valueAsDouble;
+        } catch (JsonParseException e) {
+            throw createInvalidBondDataException(
+                    e,
+                    "Current JSON float value '%s' is outside of bounds for the expected Bond data type 'double'.",
+                    this.parser.getValueAsString());
+        }
+    }
+
+    @Override
+    public final boolean readBool() throws IOException {
+        final JsonToken currentToken = this.advanceParserToNextToken();
+        if (currentToken == JsonToken.VALUE_FALSE) {
+            return false;
+        } else if (currentToken == JsonToken.VALUE_TRUE) {
+            return true;
+        } else {
+            throw createInvalidBondDataException(
+                    "Current JSON token is '%s' where either '%s' or '%s' for Bond 'bool' data type is expected.",
+                    currentToken,
+                    JsonToken.VALUE_FALSE,
+                    JsonToken.VALUE_TRUE);
+        }
+    }
+
+    @Override
+    public final String readString() throws IOException {
+        final JsonToken currentToken = this.advanceParserToNextToken();
+        if (currentToken == JsonToken.VALUE_STRING) {
+            final String valueAsString = this.parser.getValueAsString();
```

Comment:
~ Inline?

Reply:
Better for debugging, no runtime impact (javac is smart enough).

---
In reply to: [143861440](https://github.com/Microsoft/bond/pull/632#discussion_r143861440) [](ancestors = 143861440)